src/python/pants/backend/python/interpreter_cache.py

```diff
@@ -61,9 +63,9 @@ def pex_python_paths(cls):
     else:
       return []
 
-  def __init__(self, python_setup, python_repos, logger=None):
-    self._python_setup = python_setup
-    self._python_repos = python_repos
+  def __init__(self, python_setup=None, python_repos=None, logger=None):
+    self._python_setup = python_setup or PythonSetup.global_instance()
```

Comment:
How about making PythonInterpreterCache a subsystem that has deps on PythonSetup and PythonRepos? This is an accident waiting to happen.

Reply:
Could do. How to pass the logger in becomes a little tricky, but in practice (outside of tests) we always use a logger that is a thin wrapper around the global RunTracker, which is itself a subsystem, so...

Reply:
Perhaps just using a `logging` logger? If we don't wire up logging <-> reporting we probably should be to allow for lighter weight decentralized logging that respects pants logging config once its online and initialized.

Reply:
I think #6771 is great. That said, I was more pointing out what this says more cleanly:
```
$ git grep "import logging" src/python/ | wc -l
93
```
I suspect you'll want to wait on #6771 since you went through the effort, but a good deal of non-task code just uses logging and this could too.